commit dd962afdd4ac74b8580cec072d1abc1a888d6771 (refs/changes/57/5157/4)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-11-30T16:08:47+01:00 (10 months ago)
    
    TRITON-961 Add "flexible_disk_space" support to VMAPI
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/lib/common/validation.js b/lib/common/validation.js
index bf86045..b75ad34 100644
--- a/lib/common/validation.js
+++ b/lib/common/validation.js
@@ -1062,10 +1062,11 @@ function validatePackageValues(papi, params, errs, callback) {
                     code: 'Invalid',
                     message: err.message
                 });
-                return callback();
+                callback();
             } else {
-                return callback(err);
+                callback(err);
             }
+            return;
         }
 
         // Check that the package is compatible with this brand
@@ -1083,6 +1084,9 @@ function validatePackageValues(papi, params, errs, callback) {
             if (params[field] === undefined && pkg[field] !== undefined) {
                 if (field === 'quota') {
                     if (['bhyve', 'kvm'].indexOf(params.brand) !== -1) {
+                        if (params.brand === 'bhyve' && pkg.flexible_disk) {
+                            params.flexible_disk_size = Number(pkg.quota);
+                        }
                         params.quota = 10;
                     } else {
                         params.quota = Number(pkg.quota) / 1024;
@@ -1124,7 +1128,7 @@ function validatePackageValues(papi, params, errs, callback) {
 
         params['package'] = pkg;
 
-        return callback();
+        callback();
     });
 }
 // export for tests
diff --git a/lib/common/vm-common.js b/lib/common/vm-common.js
index e08594f..32e1c98 100644
--- a/lib/common/vm-common.js
+++ b/lib/common/vm-common.js
@@ -171,7 +171,8 @@ function translateVm(obj, fullObject) {
         'package_version',
         'pid',
         'tmpfs',
-        'zfs_data_compression'
+        'zfs_data_compression',
+        'flexible_disk_size'
     ];
 
     optionalFields.forEach(function (field) {
diff --git a/lib/workflows/provision.js b/lib/workflows/provision.js
index 17ccdb2..c848fb0 100644
--- a/lib/workflows/provision.js
+++ b/lib/workflows/provision.js
@@ -23,7 +23,7 @@ var nfsVolumes = require('./nfs-volumes');
 
 var wfapiUrl;
 
-var VERSION = '8.0.2';
+var VERSION = '8.1.2';
 
 
 /*
@@ -294,8 +294,12 @@ function preparePayload(job, cb) {
 
     if (['bhyve', 'kvm'].indexOf(params['brand']) !== -1) {
         payload.disks = params.disks;
+        var otherProps = ['disk_driver', 'nic_driver', 'cpu_type'];
+        if (params['brand'] === 'bhyve') {
+            otherProps.push('flexible_disk_size');
+        }
 
-        ['disk_driver', 'nic_driver', 'cpu_type'].forEach(function (field) {
+        otherProps.forEach(function addOtherPropsToPayload(field) {
             if (params[field]) {
                 payload[field] = params[field];
             } else {
